consul/base.py

```diff
@@ -2440,3 +2540,35 @@ def raft_config(self):
             """
             return self.agent.http.get(
                 CB.json(), '/v1/operator/raft/configuration')
+
+    class Connect:
+        def __init__(self, agent):
+            self.agent = agent
+            self.ca = Consul.Connect.CA(agent)
+
+        class CA:
+            def __init__(self, agent):
+                self.agent = agent
+
+            def roots(self, pem=False, token=None):
+                params = []
+                params.append(('pem', int(pem)))
+                token = token or self.agent.token
+                if token:
+                    params.append(('token', token))
+
+                return self.agent.http.get(
+                    CB.bool(),
+                    '/v1/connect/ca/roots',
```

Comment:
Use global var for '/v1/connect/ca/' as used multiple times?

Reply:
It was not the spirit of this codebase for existing features. We could even, with the naming of classes and functions deduce the URI, but I kept it aligned with the code in place.
This will require dedicated code-style/refactoring reviews.

Reply:
Indeed - let's keep this as it is for now then. FAR